src/pod/personal-storage.ts

```diff
@@ -104,4 +107,31 @@ export class PersonalStorage {
       podsInfo.lookupAnswer!.epoch.getNextEpoch(getUnixTimestamp()),
     )
   }
+
+  /**
+   * Shares pod information
+   *
+   * @param name pod name
+   *
+   * @returns swarm reference of shared metadata about pod
+   */
+  async share(name: string): Promise<Reference> {
+    assertActiveAccount(this.accountData)
+    assertPodName(name)
+    const wallet = this.accountData.wallet!
+    const podInfo = await getExtendedPodsList(
+      this.accountData.connection.bee,
+      name,
+      wallet,
+      this.accountData.connection.options?.downloadOptions,
+    )
+
+    const data = stringToBytes(
+      JSON.stringify(
+        createPodShareInfo(name, podInfo.podAddress, this.accountData.username!, prepareEthAddress(wallet.address)),
```

Comment:
I would suggest to add the Ethereum address of the user instead of `this.accountData.username!`.

We don't want to tie the user to their username and also make ENS registration requirement.

Reply:
agree